crates/ironrdp-rdpdr-native/src/nix/simple.rs

```diff
@@ -0,0 +1,765 @@
+use std::os::unix::fs::MetadataExt;
+use tracing::{debug, warn};
+
+#[derive(Debug)]
+pub struct SimpleRdpdrBackend {
+    pub file_id: u32,
+    pub file_base: String,
+    pub file_map: std::collections::HashMap<u32, std::fs::File>,
+    pub file_path_map: std::collections::HashMap<u32, String>,
+    pub file_dir_map: std::collections::HashMap<u32, DirBox>,
+}
+#[derive(Debug)]
+pub struct DirBox {
+    pub dir: *mut DIR,
+}
+unsafe impl Send for DirBox {}
+
+impl_as_any!(SimpleRdpdrBackend);
+
+impl RdpdrBackend for SimpleRdpdrBackend {
+    fn handle_server_device_announce_response(&mut self, _pdu: ServerDeviceAnnounceResponse) -> PduResult<()> {
+        Ok(())
+    }
+    fn handle_scard_call(&mut self, _req: DeviceControlRequest<ScardIoCtlCode>, _call: ScardCall) -> PduResult<()> {
+        Ok(())
+    }
+    fn handle_drive_io_request(&mut self, req: ServerDriveIoRequest) -> PduResult<Vec<SvcMessage>> {
+        debug!("handle_drive_io_request:{:?}", req);
+        match req {
+            ServerDriveIoRequest::DeviceWriteRequest(req_inner) => write_device(self, req_inner),
+            ServerDriveIoRequest::ServerCreateDriveRequest(req_inner) => create_drive(self, req_inner),
+            ServerDriveIoRequest::DeviceReadRequest(req_inner) => read_device(self, req_inner),
+            ServerDriveIoRequest::DeviceCloseRequest(req_inner) => close_device(self, req_inner),
+            ServerDriveIoRequest::ServerDriveNotifyChangeDirectoryRequest(_) => {
+                // TODO
+                Ok(Vec::new())
+            }
+            ServerDriveIoRequest::ServerDriveQueryDirectoryRequest(req_inner) => query_directory(self, req_inner),
+            ServerDriveIoRequest::ServerDriveQueryInformationRequest(req_inner) => query_information(self, req_inner),
+            ServerDriveIoRequest::ServerDriveQueryVolumeInformationRequest(req_inner) => {
+                query_volume_information(self, req_inner)
+            }
+            ServerDriveIoRequest::ServerDriveSetInformationRequest(req_inner) => set_information(self, req_inner),
+            ServerDriveIoRequest::DeviceControlRequest(req_inner) => Ok(vec![SvcMessage::from(
+                RdpdrPdu::DeviceControlResponse(DeviceControlResponse {
+                    device_io_reply: DeviceIoResponse::new(req_inner.header, NtStatus::SUCCESS),
+                    output_buffer: None,
+                }),
+            )]),
+            ServerDriveIoRequest::ServerDriveLockControlRequest(_) => {
+                // TODO
+                Ok(Vec::new())
+            }
+        }
+    }
+}
+
+pub(crate) fn write_device(
+    backend: &mut SimpleRdpdrBackend,
+    req_inner: DeviceWriteRequest,
+) -> PduResult<Vec<SvcMessage>> {
+    let write_inner = |file: &mut std::fs::File, offset, write_data| -> std::io::Result<usize> {
+        let sf = SeekFrom::Start(offset);
+        file.seek(sf)?;
+        let length = file.write(write_data)?;
+        file.flush()?;
+        Ok(length)
+    };
```

Comment:
style: If capturing environment is not required, [this is the style we follow](https://github.com/Devolutions/IronRDP/blob/abc0ee0617fcdcda3f0ace3ffd292251979d1910/STYLE.md#local-helper-functions)

Reply:
ok, i modify according to our style